subworkflows/local/preparegenome.nf

```diff
@@ -72,22 +72,24 @@ workflow PREPARE_GENOME {
     /*
      * Calculate effective genome sizes
      */
-    gs = ch_fasta.size() * 0.78
+    gs = 0
     if (params.macs_gsize) {
         gs = params.macs_gsize
     } else {
-        //genome size remove all N then * 78%
-        gs = 0.0
-        ch_fasta.withReader{
-            String line
-            while( line = it.readLine() ){
-                if( !(line ==~ /^>/) ){
-                    l = line.toLowerCase()
-                    gs += l.count("a") + l.count("c") + l.count("g") + l.count("t")
+        //genome size remove all N then * 78% ref:https://github.com/macs3-project/MACS/issues/299
+        gs = ch_fasta.map{
+            gs = 0.0
+            it.withReader{
+                String line
+                while( line = it.readLine() ){
+                    if( !(line ==~ /^>/) ){
+                        l = line.toLowerCase()
+                        gs += l.count("a") + l.count("c") + l.count("g") + l.count("t")
+                    }
```

Comment:
```suggestion
        //genome size remove all N then * 78% ref:https://github.com/macs3-project/MACS/issues/299
        genome_size = ch_fasta.splitFasta( record: [id: true, seqString: true] )
        .map { it.seqString.replaceAll('[^acgtACGT]','').length() }
        .sum()
```
If you use `splitFasta` you also do not need to unzip the fasta first.
According to your reference `https://github.com/macs3-project/MACS/issues/299`, the multiplier you're using (0.78) is specific to that case. 
Which definition of effective genome size are you using? https://deeptools.readthedocs.io/en/latest/content/feature/effectiveGenomeSize.html

Reply:
I will simply using the method 1 here. Thank you for this.